### PR TITLE
using historical data base

### LINKED ISSUE #4322: StructMech Explicit Time Int. "AdvanceInTime"
Hi I need the explicit time integration to also use "AdvanceInTime" or a similar option to use it in a strong coupling algorithm. 
At the moment the strategy calls `SolveSolutionStep`, which then calls `Update` in the scheme. Which always proceeds in time. For a stroung coupling algorithm I would need a possibility to repeat a certain time step several times. 
I'll work on this now, but I thought you might already have good ideas on how to realize this.

Klaus

### PR DESCRIPTION
**Pull request by KlausBSautter**
Hi w.r.t. #4322 this should make it possible to stay in 1 time step without proceeding (e.g. for strong coupling algorithms), when using the explicit time integration.

Simple example, where I stop calling "AdvanceInTime" after 0.004 seconds. The x-axis is a custom time 

figure_1

closes #4322

### CODE REVIEW

**Comment by KlausBSautter** [comment]
makes some tests fail, I have to check this before merging !

**Comment by KlausBSautter** [comment]
OK I tested this on all of my examples and everything gives the same results. The only problem is the beginning of the simulation. Especially `Initialize` in the scheme is weird to me. In the patch tests the `mTime` variables are nonzero here but in normal cases they are zero. I have to talk to @philbucher  about that.
At the moment in the patch tests, in the first two time steps middle,previous and and current are the same where they should not be the same. So this is also fixed in this pr. 
But this leads to the fact that we have to change the results in the patch tests

**Comment by KlausBSautter** [comment]
@philbucher this should be it
@loumalouomega all the tests run. Can you check if your things still work?
thanks

File: applications/StructuralMechanicsApplication/tests/test_patch_test_truss.py
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/test_patch_test_truss.py]:
```suggestion
        if not linear_flag:
```
more pythonic

File: applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp]:
```suggestion
        mTime.Middle   = mTime.Current - 0.50*mTime.Delta;
```

please don't leave commented code in master, this is very confusing for someone else trying to understand whats going on

**Review by philbucher — COMMENTED** [review]
is this consistent now with the implicit schemes?

Regarding the tests, I guess you know what you are doing

please wait for @loumalouomega to give green light if his things are working

**Review by loumalouomega — DISMISSED** [review]
OK from my side

**Comment by KlausBSautter** [comment]
@philbucher I got rid of the comment. Can you approve?

**Review by philbucher — APPROVED** [review]

**Pull request merged**
